monai/apps/detection/networks/retinanet_detector.py

```diff
@@ -0,0 +1,896 @@
+                feature_map_scales=(1, ), base_anchor_shapes=((8,) * spatial_dims)
+            )
+            net = NaiveNet(spatial_dims, num_classes)
+            detector = RetinaNetDetector(net, anchor_generator)
+
+            # only detector.network may contain trainable parameters.
+            optimizer = torch.optim.SGD(
+                detector.network.parameters(),
+                1e-3,
+                momentum=0.9,
+                weight_decay=3e-5,
+                nesterov=True,
+            )
+            torch.save(detector.network.state_dict(), 'model.pt')  # save model
+            detector.network.load_state_dict(torch.load('model.pt'))  # load model
+    """
+
+    def __init__(
+        self,
+        network,
+        anchor_generator: AnchorGenerator,
+        box_overlap_metric: Callable = box_utils.box_iou,
+        debug: bool = False,
+        **kwargs,
+    ):
+        super().__init__()
+
+        if not all(
+            hasattr(network, attr)
+            for attr in ["spatial_dims", "num_classes", "cls_key", "box_reg_key", "num_anchors", "size_divisible"]
+        ):
+            raise AttributeError(
+                "network should have attributes, including: "
+                "'spatial_dims', 'num_classes', 'cls_key', 'box_reg_key', 'num_anchors', 'size_divisible'."
+            )
+
+        self.network = network
+        self.spatial_dims = self.network.spatial_dims
+        self.num_classes = self.network.num_classes
+        self.size_divisible = ensure_tuple_rep(self.network.size_divisible, self.spatial_dims)
+        # keys for the network output
+        self.cls_key = self.network.cls_key
+        self.box_reg_key = self.network.box_reg_key
+
+        # check if anchor_generator matches with network
+        self.anchor_generator = anchor_generator
+        self.num_anchors_per_loc = self.anchor_generator.num_anchors_per_location()[0]
+        if self.num_anchors_per_loc != self.network.num_anchors:
+            raise ValueError(
+                f"Number of feature map channels ({self.network.num_anchors}) "
+                f"should match with number of anchors at each location ({self.num_anchors_per_loc})."
+            )
+
+        self.box_overlap_metric = box_overlap_metric
+        self.debug = debug
+
+        # default setting for both training and inference
+        self.box_coder = BoxCoder(weights=(1.0,) * 2 * self.spatial_dims)
+
+        # default keys in the ground truth targets and predicted boxes
+        self.target_box_key = "boxes"
+        self.target_label_key = "labels"
+        self.pred_score_key = self.target_label_key + "_scores"  # score key for the detected boxes
+
+        # default setting for inference
+        self.use_inferer = False
+        self.score_thresh = 0.5
+        self.topk_candidates = 1000
+        self.nms_thresh = 0.2
+        self.detections_per_img = 300
```

Comment:
Since these can be task dependent it would make sense to make them configurable from the outside

Reply:
Yes, there is such function. I added comments